packages/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -702,6 +723,15 @@ export const useGridColumnResize = (
         }
 
         apiRef.current.updateColumns(newColumns);
+
+        newColumns.forEach((newColumn) => {
+          const width: number = newColumn.width as number;
+          apiRef.current.publishEvent('columnWidthChange', {
+            element: apiRef.current.getColumnHeaderElement(newColumn.field),
+            colDef: newColumn,
+            width,
+          });
+        });
```

Comment:
This will trigger for all columns regardless of if they have changed. You have the original columns in `columns`.

You can avoid typings like this, the inference knows it's a number:
```ts
const width: number = newColumn.width as number;
```

Reply:
We are doing something wrong this implementation is correct.

let me Explain in detail

There are two possibility of calling autosizeColumns function

1.  through doubleClick event Listener https://github.com/mui/mui-x/blob/next/packages/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx#L743  from here handleColumnSeparatorDoubleClick calls the autosizeColumns and over here it pass the exact column which is being resized 
2.  At the time of mounting when user pass autoResizeOnMount prop https://github.com/mui/mui-x/blob/next/packages/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx#L716

here each of the column is being resized on mounting

In our case to solve this bug we only need to concentrate on case 1 when the autosizeColumns is called on double click.

So on double click event from https://github.com/mui/mui-x/blob/next/packages/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx#L661

we get only one column that needs to be resized and we need not check on all the columns.

Now in case 2 when autOResizeOnMount is called in that case all the columns should be resized and we need to decide whether to publish event or not.

So from both the case we can figure it out that either all of the column will be resized or only 1 will be resized.


if we want to disable the event publish on all column resize (case 2) we just need to add one condition


```
if (userOptions?.columns) {
          newColumns.forEach((newColumn) => {
            const width = newColumn.width;
            apiRef.current.publishEvent('columnWidthChange', {
              element: apiRef.current.getColumnHeaderElement(newColumn.field),
              colDef: newColumn,
              width,
            });
          });
        }

```

Reply:
There is no need for an `if` guard, just check the array and dispatch one event for each column that actually changed. Compare `newColumn.width !== columns[index].width` in the `.forEach` loop to figure out which ones changed.

The autosize function is part of the public API and can be called programatically with any number of columns.

Reply:
Done